sklearn/linear_model/sag_fast.pyx

```diff
@@ -614,10 +614,14 @@ cdef void lagged_update(double* weights, double wscale, int xnnz,
                         last_update_ind = sample_itr - 1
                     for lagged_ind in range(sample_itr - 1,
                                    last_update_ind - 1, -1):
-                        grad_step = (cumulative_sums[lagged_ind]
-                           - cumulative_sums[lagged_ind - 1])
-                        prox_step = (cumulative_sums_prox[lagged_ind]
-                           - cumulative_sums_prox[lagged_ind - 1])
+                        if lagged_ind > 1:
```

Comment:
Yes. We choose pointers because apparently they're a little more efficient than memoryviews, and much better where we want to handle read-only memmaps. But they're much less safe!